### DISCUSSION TITLE
Dynamic position for dynamic number of handle

### DISCUSSION THREAD

**Discussion by Ipicky22** [xyflow/xyflow#2283] [Ideas]
Hi ! 

I try to create the most generic, as possible, custom node.
Its structure is based on json file.
For the handle source or target, my json have two array of object, for example like that :
```json
"inputs" : [
{
"name": "input-1",
"type": "string",
"description": "input-1-description"
}
],
"outputs" : [
{
"name": "output-1",
"type": "string",
"description": "output-1-description"
}
],
```

So i create a function that map my array and return Handle Component.
The problem is that the position is define by ```position={Position.Left}``` for example, and the handle is automatically center.

Is it possible to create, in React-Flow, a good distribution of handles on the line ?

I already try, to use ```useRef``` on my div container, to catch height and width, then calculate spacing between handle, and add it in style, but doesn't work.

If anyone have a solution, I take it !

**Comment by bowmanconnor** [comment]
Any solution? I am also trying to set the location of a handle to not be in the center.

**Comment by moklick** [comment]
Hey @Ipicky22 

that's possible. You need to overwrite some default handle styles to achieve this. I created a simplified demo that shows ho to do this: https://codesandbox.io/s/sandpack-project-forked-u6tx30?file=/HandlesNode.js

**Reply by Ipicky22** [reply]
Sorry for the late answer, I will try it ! Thanks a lot.

**Reply by rodripf** [reply]
Hello @moklick!
In the solution you provided, when you try to draw an edge, the starting point of the edge is misplaced. Could you please provide a hint on how to fix this behaviour?
Thank you!

**Reply by bcakmakoglu** [reply]
The problem in that old sandbox is that `id={`i`}` results in an undefined id.
Just pass a proper id per handle and you're good to go: https://codesandbox.io/p/sandbox/sandpack-project-forked-jqpvy4?file=%2FHandlesNode.js%3A34%2C19
